### PR TITLE
Update to JUnit5

### PR DESCRIPTION
**Pull request by bryanlb**
Updates test suite to JUnit 5. Additionally refactors a handful of tests that had low hanging fruit for runtime optimizations, resulting in a combined execution time reduction of approximately 110s.

### CODE REVIEW

**Comment by vthacker** [comment]
https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4-tips + Ignore now becoming Disabled is a good guide to review the chances

File: kaldb/src/test/java/com/slack/kaldb/blobfs/LocalBlobFsTest.java [hunk 1]
```diff
@@ -59,144 +62,142 @@ public void testFS() throws Exception {
...
-    Assert.assertFalse(nonExistingFile.exists());
+    assertFalse(nonExistingFile.exists());
     localBlobFs.touch(nonExistingFile.toURI());
-    Assert.assertTrue(nonExistingFile.exists());
+    assertTrue(nonExistingFile.exists());
     long currentTime = System.currentTimeMillis();
-    Assert.assertTrue(localBlobFs.lastModified(nonExistingFile.toURI()) <= currentTime);
-    Thread.sleep(1000L);
+    assertTrue(localBlobFs.lastModified(nonExistingFile.toURI()) <= currentTime);
+    Thread.sleep(1L);
```
Comment (vthacker) [inline kaldb/src/test/java/com/slack/kaldb/blobfs/LocalBlobFsTest.java:191]: will this be enough?

File: kaldb/src/test/java/com/slack/kaldb/chunk/ChunkInfoTest.java [hunk 2]
```diff
@@ -253,54 +254,62 @@ public void testChunkDataInRange() {
         .isTrue();
   }
 
-  @Test(expected = IllegalArgumentException.class)
+  @Test
```
[the remark this answers is not in the document]
Comment (bryanlb) [reply kaldb/src/test/java/com/slack/kaldb/chunk/ChunkInfoTest.java:257]: One of the main reasons for getting rid of this syntax is any of the code can throw the exception and "pass" the test. By moving to assert wrappers you can guarantee that specific invocations are what is doing the throwing. 🧠

File: kaldb/src/test/java/com/slack/kaldb/chunkManager/ChunkCleanerServiceTest.java [hunk 3]
```diff
@@ -197,7 +201,7 @@ private void testBasicSnapshotMetadata(Instant creationTime) {
     assertThat(snapshotNodes.get(0).partitionId).isEqualTo(TEST_KAFKA_PARTITION_ID);
     assertThat(snapshotNodes.get(0).snapshotId).startsWith(SnapshotMetadata.LIVE_SNAPSHOT_PATH);
     assertThat(snapshotNodes.get(0).startTimeEpochMs)
-        .isCloseTo(creationTime.toEpochMilli(), Offset.offset(1000L));
+        .isCloseTo(creationTime.toEpochMilli(), Offset.offset(2500L));
```
Comment (vthacker) [inline kaldb/src/test/java/com/slack/kaldb/chunkManager/ChunkCleanerServiceTest.java:204]: this will make it less flaky?

**Review by vthacker — APPROVED** [review]
let's go!

File: kaldb/src/test/java/com/slack/kaldb/chunkManager/ChunkCleanerServiceTest.java (hunk 3 above)
Comment (bryanlb) [reply kaldb/src/test/java/com/slack/kaldb/chunkManager/ChunkCleanerServiceTest.java:204]: It should - I was having some intermittent time issues as well running this locally, as 1s appeared to be a bit too strict.

**Pull request merged**